unit_tests/test_redis_aggregator.py

```diff
@@ -62,44 +96,98 @@ def fixture_mock_client_command_buffer(mocker):
+        assert agg._client_command_buffer is client_command_buffer_mock
 
     @staticmethod
-    def test_constructor_threadpoolexecutor_instantiated(aggregator):
-        assert aggregator.executor is not None
+    @pytest.mark.usefixtures("mock_transaction_id_and_timeout_blocking")
+    def test_constructor_threadpoolexecutor_instantiated():
+        ThreadPoolExecutor_mock = MagicMock
+        with patch("gsy_e_sdk.redis_aggregator.ThreadPoolExecutor",
+                   return_value=ThreadPoolExecutor_mock) as mocked_class:
+            agg = RedisAggregator(aggregator_name=TEST_AGGREGATOR_NAME)
+            mocked_class.assert_called()
+        assert agg.executor is ThreadPoolExecutor_mock
 
     @staticmethod
-    def test_constructor_connect_and_subscribe_called():
-        with patch("gsy_e_sdk.redis_aggregator."
-                   "RedisAggregator._connect_and_subscribe") as mocked_method:
-            RedisAggregator(aggregator_name=TEST_AGGREGATOR_NAME)
-            mocked_method.assert_called()
+    @pytest.mark.usefixtures("mock_transaction_id_and_timeout_blocking")
+    def test_constructor_connect_and_subscribe_side_effects_1():
+        """Test the side effects due to the methods called
+        with _connect_and_subscribe()"""
+
+        # side effects of _subscribe_to_aggregator_response_and_start_redis_thread
+        # private method
+        aggregator = RedisAggregator(aggregator_name=TEST_AGGREGATOR_NAME)
+
+        channel_dict_1 = {"aggregator_response": aggregator._aggregator_response_callback}
+        aggregator.pubsub.run_in_thread.assert_called_with(daemon=True)
+
+        # side effects of _connect_to_simulation -> _create_aggregator private method
+        data = {"name": aggregator.aggregator_name, "type": "CREATE",
+                "transaction_id": TEST_TRANSACTION_ID}
+        aggregator.redis_db.publish.assert_called_with("aggregator", json.dumps(data))
+        assert TEST_TRANSACTION_ID in aggregator._transaction_id_buffer
+        assert aggregator.aggregator_uuid is TEST_TRANSACTION_ID
+
+        # side effects of _subscribe_to_response_channels private method
+        channel_dict_2 = {f"external-aggregator/*/{aggregator.aggregator_uuid}/events"
+                          f"/all": aggregator._events_callback_dict,
+                          f"external-aggregator/*/{aggregator.aggregator_uuid}/response"
+                          f"/batch_commands": aggregator._batch_response,
+                          }
+
+        aggregator.pubsub.psubscribe.assert_has_calls([call(**channel_dict_1),
+                                                       call(**channel_dict_2)])
 
     @staticmethod
-    def test_constructor_lock_instantiated(aggregator):
-        assert aggregator.lock is not None
+    @pytest.mark.usefixtures("mock_transaction_id_and_timeout_blocking")
+    def test_constructor_lock_instantiated():
+        with patch("gsy_e_sdk.redis_aggregator.Lock"
+                   ) as mocked_class:
```

Comment:
Why aren't you mocking the exact `Lock` instance as you've done in the other tests? That would end up in an assertion like
```python
assert aggregator.lock is lock_mock
```

Reply:
I was asking myself exactly the same. There is not reason. Probably I just missed it.